styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/ExecutionDescriptionHandler.java

```diff
@@ -72,8 +82,9 @@ public void transitionInto(RunState state, EventRouter eventRouter) {
     switch (state.state()) {
       case PREPARE:
         try {
+          final String executionId = createExecutionId();
           final Event submitEvent = Event.submit(
-              state.workflowInstance(), getExecDescription(workflowInstance, state.data()), createExecutionId());
+              state.workflowInstance(), getExecDescription(workflowInstance, state.data(), executionId), executionId);
```

Comment:
can't we create the flyteExecId the moment we create the executionId? 
here we pass the executionId only to create the FlyteExecutionId

Reply:
We can do that as well. But in the current way, flyteExecutionId is only calculated when there is a need for that.

Reply:
Changed to pass flyte execution id since I think it improves readability.